### PR TITLE
Count of conf participants is echoed to caller upon entry

### PR DESCRIPTION
**Pull request by danielfinke**
Lets the conference caller know if anybody is already waiting in the conference when they arrive, and how many

### CODE REVIEW

**Comment by lazedo** [comment]
@danielfinke can you put this as optional ?

File: applications/conference/src/conf_discovery_req.erl [hunk 1]
```diff
@@ -181,6 +182,22 @@ handle_search_error(Conference, Call, Srv) ->
...
     end.
 
+-spec play_participants_count(whapps_call:call(), non_neg_integer() | wh_json:object()) -> 'ok'.
+play_participants_count(Call, 0) ->
+    whapps_call_command:prompt(<<"conf-alone">>, Call),
+    'ok';
+play_participants_count(Call, 1) ->
+    whapps_call_command:prompt(<<"conf-single">>, Call),
+    'ok';
+play_participants_count(Call, Count) when is_integer(Count) andalso Count > 0 ->
```
Comment (lazedo) [inline applications/conference/src/conf_discovery_req.erl]: @danielfinke don't be defensive here `andalso Count > 0 `

**Comment by danielfinke** [comment]
Ready to go now I think with the optional setting

File: applications/conference/src/conf_discovery_req.erl [hunk 2]
```diff
@@ -181,6 +182,29 @@ handle_search_error(Conference, Call, Srv) ->
             handle_resource_locked(Conference, Call, Srv)
     end.
 
+-spec maybe_play_participants_count(whapps_conference:conference(), whapps_call:call(), non_neg_integer() | wh_json:object()) -> 'ok'.
```
Comment (lazedo) [inline applications/conference/src/conf_discovery_req.erl]: no need for wh_json:object() here

File: applications/conference/src/conf_discovery_req.erl [hunk 3]
```diff
@@ -181,6 +182,29 @@ handle_search_error(Conference, Call, Srv) ->
...
+play_participants_count(Call, 1) ->
+    whapps_call_command:prompt(<<"conf-single">>, Call),
+    'ok';
+play_participants_count(Call, Count) when is_integer(Count) ->
+    whapps_call_command:audio_macro([{'prompt', <<"conf-there_are">>}
+                                     ,{'say', wh_util:to_binary(Count), <<"number">>}
+                                     ,{'prompt', <<"conf-other_participants">>}
+                                    ], Call),
+    'ok';
+play_participants_count(Call, JObj) ->
```
Comment (lazedo) [inline applications/conference/src/conf_discovery_req.erl]: function clause not needed

File: applications/conference/src/conf_discovery_req.erl [hunk 4]
```diff
@@ -181,6 +182,29 @@ handle_search_error(Conference, Call, Srv) ->
...
     end.
 
+-spec maybe_play_participants_count(whapps_conference:conference(), whapps_call:call(), non_neg_integer() | wh_json:object()) -> 'ok'.
+maybe_play_participants_count(Conference, Call, CountData) ->
+    case whapps_conference:play_participants_count(Conference) of
+        'true' -> play_participants_count(Call, CountData);
+        'false' -> 'ok'
+    end.
+
+-spec play_participants_count(whapps_call:call(), non_neg_integer() | wh_json:object()) -> 'ok'.
```
Comment (lazedo) [inline applications/conference/src/conf_discovery_req.erl]: same here

**Comment by lazedo** [comment]
@danielfinke please rebase and apply the recommended changes. Thank You

**Comment by lazedo** [comment]
@danielfinke any update ?

**Comment by jamesaimonetti** [comment]
@danielfinke please re-open when you're ready to go on this again.

Thanks!

**Pull request closed without merging**